assertion/function/preprocess/cfg.go

```diff
@@ -119,6 +124,31 @@ func copyGraph(graph *cfg.CFG) *cfg.CFG {
 	return newGraph
 }
 
+// fixNoReturnBlock trims the `Succs` fields of a block if it contains a call to a trusted function
+// that does not return (e.g., `os.Exit`, `log.Fatal`). This ensures that our further analysis
+// correctly understands that the code is unreachable after such a call.
+func (p *Preprocessor) fixNoReturnBlock(block *cfg.Block) {
+	// No need to fix empty blocks or blocks that already have no successor.
+	if len(block.Nodes) == 0 || len(block.Succs) == 0 {
+		return
+	}
+
+	for _, node := range block.Nodes {
+		stmt, ok := node.(*ast.ExprStmt)
+		if !ok {
+			continue
+		}
+		call, ok := stmt.X.(*ast.CallExpr)
+		if !ok {
+			continue
+		}
+		if trustedfunc.TrimSuccsOn(p.pass, call) {
+			block.Succs = nil
```

Comment:
I was just wondering if we should set successors to nil or point to the return block... What is the behavior of the drivers where this behavior is handled correctly? Perhaps we can simulate that behavior here.

Reply:
I compared the correct behaviors and it seems that `block.Succs == nil` is the behavior, so we're following that practice.

This is also documented: https://pkg.go.dev/golang.org/x/tools/go/cfg

> A block may have 0-2 successors: zero for a return block or a block that calls a function such as panic that never returns; one for a normal (jump) block; and two for a conditional (if) block.

Anyway, I think this comment is no longer applicable since we decided to just fix the upstream `ctrlflow` analyzer instead of doing hacky postfixes in NilAway.

Reply:
Got it. Makes sense. Thanks for clarifying :)